### PR TITLE
Feature/hash check endpoint

### LINKED ISSUE #92: Add backend route to verify hash of a course unit
Upon receiving the hash of a course unit and its id it must verify it is the newest version or not and return the correct one.

- The endpoint should be `GET`
- It should receive a list of (course_unit_id, hash)
- It should return a list of (course_unit_id, valid)

### PR DESCRIPTION
**Pull request by jose-carlos-sousa**
closes #92 
As requested in the issue description, I implemented:

- GET hashCheck/ endpoint
- It accepts the data type Form Data as suggested by @tomaspalma 
- Request Body should have pairs key-value of course_unit_id-hash
- I return in the response body array of pairs course_unit_id-valid

### CODE REVIEW

File: mysql/sql/00_schema_mysql.sql [hunk 1]
```diff
@@ -95,29 +96,33 @@ CREATE TABLE `class` (
 --
 
 CREATE TABLE `slot` (
-  `id` INTEGER PRIMARY KEY,
+  `id` INTEGER AUTO_INCREMENT PRIMARY KEY,
```
Comment (tomaspalma) [inline mysql/sql/00_schema_mysql.sql:99]: Why did you add the `AUTO_INCREMENT` field?

It should not be there, since the slot ids are inserted via the scrapper

File: mysql/sql/00_schema_mysql.sql [hunk 2]
```diff
@@ -95,29 +96,33 @@ CREATE TABLE `class` (
...
   `lesson_type` varchar(3) NOT NULL,
   `day` tinyint(3) NOT NULL,
   `start_time` decimal(3,1) NOT NULL,
   `duration` decimal(3,1) NOT NULL,
   `location` varchar(31) NOT NULL,
   `is_composed` boolean NOT NULL,
   `professor_id` int (11),
-  `last_updated` datetime NOT NULL
+  `class_id` int(11) NOT NULL,
```
Comment (tomaspalma) [inline mysql/sql/00_schema_mysql.sql:107]: There should not any `class_id`. This was already removed in another inside the `develop` branch, was this a merge conflict gone wrong?

File: django/statistics.sql
Comment (tomaspalma) [inline django/statistics.sql:1]: This file should not be here.

Please remove it using `git rm --cached django/statistics.sql' and then push the changes

File: mysql/sql/01_data_new.sql
Comment (tomaspalma) [inline mysql/sql/01_data_new.sql:1]: The data file should not be commited as well.

Please remove it using `git rm --cached 01_data_new.sql` in your local terminal

File: mysql/sql/00_schema_mysql.sql [hunk 3]
```diff
@@ -156,6 +161,18 @@ CREATE TABLE `info` (
...
+
+CREATE TABLE `statistics` (
+  `int` INTEGER AUTO_INCREMENT PRIMARY KEY,
+  `course_unit_id` int(11) NOT NULL,
+  `acronym` varchar(10) NOT NULL,
+  `visited_times` int(11) NOT NULL,
+  `last_updated` datetime NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+
+
```
Comment (tomaspalma) [inline mysql/sql/00_schema_mysql.sql:175]: This should not be here, was this a merge conflict gone wrong?

File: mysql/sql/00_schema_mysql.sql [hunk 4]
```diff
@@ -178,3 +195,13 @@ CREATE INDEX `course_metadata_index` ON `course_metadata` (`course_id`, `course_
...
+--
+-- Indexes for table `slot`
+--
+CREATE INDEX `slot_course_unit_id` ON `slot` (`class_id`);
+
+--
+-- Indexes for table `statistics`
+--
+CREATE INDEX `statistics` ON `statistics` (`course_unit_id`);
```
Comment (tomaspalma) [inline mysql/sql/00_schema_mysql.sql:207]: This should not be here, was this a merge conflict gone wrong?

**Review by tomaspalma — CHANGES_REQUESTED** [review]

File: django/university/tests.py [hunk 5]
```diff
@@ -1,4 +1,13 @@
...
+print("olá")
+
+
+#teste de hash
+print("A hash está correta???")
+hash_check=True
+if hash_check:
+    print(" PASS")
+else:
+    print(" FAIL")
```
Comment (tomaspalma) [inline django/university/tests.py:13]: Is this necessary?

**Review by tomaspalma — CHANGES_REQUESTED** [review]

File: django/university/urls.py [hunk 6]
```diff
@@ -9,6 +9,7 @@
     path('course_unit/<int:course_unit_id>/', views.course_unit_by_id),
     path('class/<int:course_unit_id>/', views.classes),
     path('professors/<int:slot>/', views.professor),
-    path('info/', views.info)
+    path('info/', views.info),
+    path('checkHashes/', views.verify_course_units)
```
Comment (tomaspalma) [inline django/university/urls.py:13]: Very good work!

Sorry for the nitpicking, but maybe the endpoint should not contain any implementation det ail (in order for it not being necessary to change the name of the endpoint if for some reason we change the logic inside of the django view).

Maybe `course_unit/validate`?

Comment (jose-carlos-sousa) [reply django/university/urls.py]: Your suggestion makes sense and as it follows the convention resource/action it's also more meaningful and more consistent with the previous ones

File: django/university/urls.py [hunk 7]
```diff
@@ -9,6 +9,7 @@
     path('course_unit/<int:course_unit_id>/', views.course_unit_by_id),
     path('class/<int:course_unit_id>/', views.classes),
     path('professors/<int:slot>/', views.professor),
-    path('info/', views.info)
+    path('info/', views.info),
+    path('course_unit/validate', views.verify_course_units)
```
[the remark this answers is not in the document]
Comment (jose-carlos-sousa) [reply django/university/urls.py]: Yes now that the function of the endpoint changed that name makes more sense, thanks for pointing that out

**Review by tomaspalma — APPROVED** [review]
LGTM :rocket:
